# SPDX-License-Identifier: GPL-3.0-only
# eurorack f4cfa7697095a28d292bff758beb51deb49b5a86; preserve the 32-bit exponent adjustment.
--- a/plaits/dsp/fm/dx_units.h
+++ b/plaits/dsp/fm/dx_units.h
@@ -76,7 +76,7 @@
   } else if (order == 3) {
     r.f = 1.0f + x * (0.6958f + x * (0.2251f + x * 0.0791f));
   }
-  r.w += x_integral << 23;
+  r.w += static_cast<int32_t>(static_cast<uint32_t>(x_integral) << 23);
   return r.f;
 }
 
